explainability/explainability-service/src/main/java/org/kie/kogito/explainability/ExplanationServiceImpl.java

```
@@ -61,17 +63,17 @@ public ExplanationServiceImpl(
                 request.getExecutionId(),
                 request.getModelIdentifier().getResourceType(),
                 request.getModelIdentifier().getResourceId());
-        Prediction prediction = getPrediction(request.getInputs(), request.getOutputs());
-        return localExplainer.explainAsync(prediction, predictionProvider)
-                .thenApply(input -> createResultDto(input, request.getExecutionId()))
-                .exceptionally(throwable -> {
-                    LOG.error("Exception thrown during explainAsync", throwable);
-                    return new ExplainabilityResultDto(request.getExecutionId(), Collections.emptyMap());
-                });
+        try {
+            return localExplainer.explainAsync(getPrediction(request.getInputs(), request.getOutputs()), predictionProvider)
+                    .thenApply(input -> createSucceededResultDto(request.getExecutionId(), input))
+                    .exceptionally(e -> createFailedResultDto(request.getExecutionId(), e));
+        } catch (Exception e) {
```

Is the try catch needed in this case? Is there a test for such scenario?

---

Yes it is needed and there is a test. Please check [this discussion](https://github.com/kiegroup/kogito-apps/pull/428#discussion_r482002768).

---

This catch is to prevent `explainAsync` method to throw an unexpected exception instead of return the `CompletableFuture`